### PR TITLE
Dataplex JUnits

### PR DESCRIPTION
**Pull request by Shubhangi-cs**

### CODE REVIEW

File: src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java [hunk 1]
```diff
@@ -28,26 +33,39 @@
...
+import org.mockito.internal.util.reflection.FieldSetter;
 import org.powermock.api.mockito.PowerMockito;
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.PowerMockRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java]: Avoid Static imports.

File: src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java [hunk 2]
```diff
@@ -576,4 +615,729 @@ public void validateFormatForStorageBucketWhenFormatIsNotNull() {
...
+    io.cdap.cdap.api.data.schema.Schema schema = io.cdap.cdap.api.data.schema.Schema.recordOf("record",
+      io.cdap.cdap.api.data.schema.Schema.Field.of("order",
+        io.cdap.cdap.api.data.schema.Schema.of(io.cdap.cdap.api.data.schema.Schema.Type.LONG)));
+    MockFailureCollector mockFailureCollector = new MockFailureCollector();
+    GCPConnectorConfig dataplexConnectorConfig = mock(GCPConnectorConfig.class);
+    GoogleCredentials googleCredentials = Mockito.mock(GoogleCredentials.class);
+    PowerMockito.mockStatic(DataplexUtil.class);
+    DataplexServiceClient dataplexServiceClient = mock(DataplexServiceClient.class);
+    Mockito.doNothing().when(dataplexServiceClient).close();
+    when(dataplexConnectorConfig.isServiceAccountJson()).thenReturn(new Boolean(Boolean.TRUE));
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java]: `new Boolean(Boolean.TRUE)` can be simplified to just `Boolean.TRUE`

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+public class DataplexInputFormatProviderTest {
+
+  @Mock
+  JobContext mockContext;
+
+  @Mock
+  private Configuration configuration;
+
+  @Mock
+  private InputFormat inputFormat;
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java]: Unused variable?

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ConfigurationUtils.class, GCPUtils.class, GoogleCredentials.class, DataplexUtil.class})
+public class DataplexInputFormatProviderTest {
+
+  @Mock
+  JobContext mockContext;
+
+  @Mock
+  private Configuration configuration;
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java:41]: A mock object for the `Configuration` class has been created at the class level but is not being used. Instead, the mock object is created in every test case at the method level. If we aren't setting new values to it in every test case, we can skip the creation at the method level and just declare it once at the class level and use it everywhere.

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  private InputFormat inputFormat;
+
+  @Mock
+  private InputSplit inputSplit;
+
+  @Mock
+  private TaskAttemptContext context;
+
+  @Mock
+  private ConfigurationUtils configurationUtils;
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java]: Unused variable?

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      "$DataplexInputFormat", dataplexInputFormatProvider.getInputFormatClassName());
+  }
+
+  @Test
+  public void testGetInputFormatDataplexClassNameWithBigquery() throws Exception {
+    Configuration configuration = mock(Configuration.class);
+    Map<String, String> inputFormatConfiguration = new HashMap<>();
+    DataplexInputFormatProvider.DataplexInputFormat dataplexInputFormat = new
+      DataplexInputFormatProvider.DataplexInputFormat();
+    inputFormatConfiguration.put("key", "value");
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java]: Are we making use of `inputFormatConfiguration` after putting values in it?

File: src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java (hunk 1 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java]: Removed

File: src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java (hunk 2 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java]: done

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java (hunk 3 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java]: removed

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java (hunk 4 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java:41]: done

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java (hunk 5 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java]: removed

File: src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java (hunk 6 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java]: Yes it was unused so removed it.

**Pull request merged**